File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
import heapq

class Solution:
    def findKthLargest(self, nums, k):
        pq = []
        n = len(nums)
        result = float("inf")
        for i in range(n):
            heapq.heappush(pq, -nums[i])

            if len(pq) > n-k:
                result = min(result, -heapq.heappop(pq))

        return result

# Max Heap
# time complexity: O(n log k)
# space complexity: O(k)

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        for(int num : nums){
            pq.add(-num);
        }
        int result = 0;

        while(k > 0){
            result = pq.poll();
            k--;
        }

        return -result;

    }
}